<launch>
  <node pkg="sensor_imu_mti_xsens" type="xsens_mti" name="xsens_mti" output="screen">

    <param name="port_name" value="/dev/ttyUSB0" />
    <param name="frequency" value="400" />
    <param name="frame_name" value="imu_xsens_position" />

    <!-- Choose if system is configurated. NoSystemConfig == 1: Preconfigured Device can be used -->
    <param name="NoSystemConfig" value="0" />

    <!-- If  NoSystemConfig == 0::    0 = Orientation, 1 = RawData (calibrated), 2 = MagneticField 3 = Orientation,Accel,Gyr,Magnetic-->
    <param name="configuration" value="0" />

    <!--From the MTI-300 manual, the drift for Yaw angle is 1deg, for Roll/Pitch 0.2deg. Time period unknown -->
    <rosparam param="orientation_covariance">[100.0, 0.0, 0.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.01]</rosparam>
  </node>

  <node pkg="imu_filter_madgwick" type="imu_filter_node" name="madgwick_imu_filter_node" output="screen">
    <param name="mag_subscribe_topic_name" value="/sensor/imu/xsens_mti/mag_calib" />
    <param name="imu_subscribe_topic_name" value="/sensor/imu/xsens_mti/data" />

    <param name="imu_publish_topic_name" value="madgwick_imu_filter_node/imu" />
    <param name="mag_publish_topic_name" value="NOTUSED" />

    <param name="use_mag" value="true" /> <!-- Use magnetometer data ?? -->
    <param name="use_magnetic_field_msg" value="true" /> <!-- is the magnetometer data in the standard message type ?? -->

    <!-- Load imu parameters -->
    <rosparam command="load" file="$(find imu_visualization)/configs/Xsens_imu_sensor_config.yaml" />
  </node>

  <node pkg="imu_visualization" type="display_3D_visualization.py" name="Filtered_data_vis_node" output="screen">
    <param name="imu_topic" value="madgwick_imu_filter_node/imu" />
    <param name="imu_name" value="Xsens_filtered" />
    <param name="mag_topic" value="/sensor/imu/xsens_mti/mag_calib" />
  </node>

  <node pkg="imu_visualization" type="display_3D_visualization.py" name="Raw_data_vis_node" output="screen">
    <param name="imu_topic" value="/sensor/imu/xsens_mti/data" />
    <param name="imu_name" value="Xsense_raw" />
    <param name="mag_topic" value="/sensor/imu/xsens_mti/mag_calib" />
  </node>

</launch>
